### DISCUSSION TITLE
About some compiler warnings

### DISCUSSION THREAD

**Discussion by ghost** [junit-team/junit-framework#3615] [General]
I have encountered situations when the compiler falsely warns about possible problems in using JUnit 5.

In 5.9.2 (haven't tested it on newer versions) there are two cases:

1. `Method 'someMethod' annotated with '@Test' should not declare parameter 'someParameter'`
occurs when the ParameterResolver implementation is declared in the superclass.
2. `SomeExtension should be registered at the class level `
in case when we have a field-declared implementation, for example, of AfterAllCallback, in a class annotated `@TestInstance(TestInstance.Lifecycle.PER_CLASS)`

Could they be improved?

**Comment by marcphilipp** [comment]
Where are you seeing these warnings? Sounds more like IDE inspections to me.

**Comment by ghost** [comment]
In IntelliJ IDEA.

![image](https://github.com/junit-team/junit5/assets/120301053/39e3e1e5-9278-4f1d-906b-fd81a4159226)

Should this come from the IDE and not from the framework or JUnit plugin?

**Comment by sbrannen** [comment]
Thank you for confirming that these warnings are from IDEA.

The warnings you are seeing are not under the control of JUnit. If you would like for the potential error detection and warnings to be improved, please raise an issue in the IDEA issue tracker.

In light of that, I am closing this discussion.

**Reply by scordio** [reply]
@wazzeps [IDEA-313866](https://youtrack.jetbrains.com/issue/IDEA-313866) is an example where I've been asking for improvements related to JUnit.

**Comment by yschimke** [comment]
Hit this also, lot of tracking bugs in jetbrains tracker already

https://youtrack.jetbrains.com/issues?q=%22should%20not%20declare%20parameter%22
